code/game/objects/items/airbubble.dm

```diff
@@ -111,6 +123,24 @@
 		queue_smooth_neighbors(src)
 	return ..()
 
+/obj/structure/closet/airbubble/open()
+	if(opened)
+		return 0
+
+	if(!can_open())
+		return 0
+
+	dump_contents()
+
+	if(!do_after(usr, 0.35 SECONDS, act_target = src))
```

Comment:
`usr` in a non-front layer function. Make `/obj/structure/closet/proc/toggle(mob/user as mob)` in closets.dm pass `user` as an argument to both `closet/proc/close` and `closet/proc/open`. You don't have to change any of the other overrides, just the initial declaration and then this.